allure-framework-modules/allure-webapi-module/src/main/java/com/capgemini/ntc/example/core/BasePage.java

```diff
@@ -0,0 +1,138 @@
+package com.capgemini.ntc.example.core;
+
+import com.capgemini.ntc.example.core.base.driver.DriverManager;
+import com.capgemini.ntc.example.core.base.properties.PropertiesFileSettings;
+import com.capgemini.ntc.example.core.base.runtime.RuntimeParameters;
+import com.capgemini.ntc.test.core.BaseTest;
+import com.capgemini.ntc.test.core.BaseTestWatcher;
+import com.capgemini.ntc.test.core.ITestObserver;
+import com.capgemini.ntc.test.core.ModuleType;
+import com.capgemini.ntc.test.core.analytics.IAnalytics;
+import com.capgemini.ntc.test.core.base.environment.IEnvironmentService;
+import com.capgemini.ntc.test.core.base.properties.PropertiesSettingsModule;
+import com.capgemini.ntc.test.core.logger.BFLogger;
+import com.google.inject.Guice;
+
+import ru.yandex.qatools.allure.annotations.Attachment;
+
+abstract public class BasePage implements ITestObserver {
+	
+	private static DriverManager driver = null;
+	
+	private final static PropertiesFileSettings propertiesFileSettings;
+	private static IEnvironmentService environmentService;
+	private final static IAnalytics analytics;
+	public final static String analitycsCategoryName = "NAME-OF-MODULE"; // Selenium-Module
+	
+	static {
+		// Get analytics instance created in BaseTets
+		analytics = BaseTest.getAnalytics();
+		
+		// Get and then set properties information from selenium.settings file
+		propertiesFileSettings = setPropertiesSettings();
+		
+		// Read System or maven parameters
+		setRuntimeParametersSelenium();
+		
+		// Read Environment variables either from environmnets.csv or any other input data.
+		setEnvironmetInstance();
+	}
+	
+	public static IAnalytics getAnalytics() {
+		return BasePage.analytics;
+	}
+	
+	public BasePage() {
+		this(getDriver());
+	}
+	
+	public BasePage(DriverManager driver) {
+		// Add given module to Test core Observable list
+		this.addObserver();
+		
+	}
+	
+	@Override
+	public void addObserver() {
+		BaseTestWatcher.addObserver(this);
+	}
+	
+	@Override
+	public void onTestFailure() {
+		BFLogger.logDebug("BasePage.onTestFailure    " + this.getClass()
+				.getSimpleName());
+		makeScreenshotOnFailure();
+		makeSourcePageOnFailure();
```

Comment:
please remove `makeScreenshotOnFailure()`  and `makeSourcePageOnFailure()`. 

Would like to do something while test fails ?